### PR TITLE
Service account resource

### PR DESCRIPTION
**Pull request by fantapop**
Added a resource for managing service accounts.

Vendor changes were put in a separate commit and are still a WIP.  That commit can be ignored. Once the necessary api changes land in production, I'll update the PR to include the correct vendor changes from the cockroach-cloud-sdk.

Also, added here is a commit to add some logging of api invocations.

**Commit checklist**
- [x] Changelog
- [x] Doc gen (`make generate`)
- [x] Integration test(s)
- [x] Acceptance test(s)
- [x] Example(s)

- - -
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202)

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -60,7 +60,7 @@ github.com/cloudflare/circl/math/mlsbset
 github.com/cloudflare/circl/sign
 github.com/cloudflare/circl/sign/ed25519
 github.com/cloudflare/circl/sign/ed448
-# github.com/cockroachdb/cockroach-cloud-sdk-go v1.9.0
+# github.com/cockroachdb/cockroach-cloud-sdk-go v1.9.0 => /Users/fitzner/go/src/github.com/cockroachdb/cockroach-cloud-sdk-go
```
Comment (fantapop) [inline vendor/modules.txt:63]: this will not be included after I update to the actual sdk version

File: vendor/modules.txt [hunk 2]
```diff
@@ -568,3 +569,4 @@ gopkg.in/yaml.v2
 # gopkg.in/yaml.v3 v3.0.1
 ## explicit
 gopkg.in/yaml.v3
+# github.com/cockroachdb/cockroach-cloud-sdk-go => /Users/fitzner/go/src/github.com/cockroachdb/cockroach-cloud-sdk-go
```
Comment (fantapop) [inline vendor/modules.txt:572]: this will not be included after I update to the actual sdk version

**Review by rgcase — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202#-:-NwGbbS3DGTiXxe-1oIW:bnfp4nl)* status: 0 of 60 files reviewed, 1 unresolved discussion (waiting on @fantapop and @pawalt)

___
*[`-- commits` line 20 at r1](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202#-NwFyqBn7_H_Q6aR1A4N):*
super nit: typo in `udpate`

**Review by fantapop — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202)* status: 0 of 60 files reviewed, 3 unresolved discussions (waiting on @pawalt and @rgcase)

___
*[`-- commits` line 20 at r1](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202#-NwFyqBn7_H_Q6aR1A4N):*
<details><summary><i>Previously, rgcase (Ryan Case) wrote…</i></summary><blockquote>

super nit: typo in `udpate`
</blockquote></details>

oh ha.  This line will get removed when I update based on the latest cloud sdk anyways.  I'll ping you to look at the updated message when that happens.

**Comment by fantapop** [comment]
@rgcase I rebased this off main and also updated it to not overwrite the description if its not set in the resource.  Let me know what you think.

**Review by rgcase — CHANGES_REQUESTED** [review]
Just a couple small comments.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202#-:-NwuGcu76tsg6WkK2AsZ:by7e4en)* status: 0 of 70 files reviewed, 4 unresolved discussions (waiting on @fantapop and @pawalt)

___
*[`internal/provider/service_account_resource.go` line 196 at r2](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202#-NwuFILR11oRXnFluFru:-NwuFILR11oRXnFluFrv:b42dpai) ([raw file](https://github.com/cockroachdb/terraform-provider-cockroach/blob/5d474ff8b129b8e2646f384bc01a326b470e582e/internal/provider/service_account_resource.go#L196)):*
> ```Go
> 
> 	newName := plan.Name.ValueString()
> 	updateSpec := &client.UpdateServiceAccountSpecification{
> ```

I think you could just do
```
updateSpec := &client.UpdateServiceAccountSpecification{
  Name := plan.Name.ValueStringPointer(),
  Description := plan.Description.ValueStringPointer()
}
```
and then optionally check if they're not both null if you want, although we're already assuming here that `plan.Name` isn't null.
___
*[`internal/provider/service_account_resource_test.go` line 101 at r2](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202#-NwuGLG18jl2fgOvcZfo:-NwuGLG18jl2fgOvcZfp:blwoouq) ([raw file](https://github.com/cockroachdb/terraform-provider-cockroach/blob/5d474ff8b129b8e2646f384bc01a326b470e582e/internal/provider/service_account_resource_test.go#L101)):*
> ```Go
> 	serviceAccountUpdated.Description = description
> 
> 	// Called by first update, description not included in resource so we don't
> ```

I don't think I understand what this comment means. Aren't we including the description in the patch?

**Review by rgcase — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/202#-:-NwuQbJsBPfM1Oa4pXBb:bnfp4nl)* status: 0 of 71 files reviewed, 3 unresolved discussions (waiting on @pawalt)

**Pull request merged**